src/Processors/Formats/Impl/NativeORCBlockInputFormat.cpp

```diff
@@ -97,6 +97,390 @@ std::unique_ptr<orc::InputStream> asORCInputStreamLoadIntoMemory(ReadBuffer & in
+                || curr.function == KeyCondition::RPNElement::FUNCTION_NOT_IN_RANGE
+                || curr.function == KeyCondition::RPNElement::FUNCTION_NOT_IN_SET;
+            const bool contains_is_null = curr.function == KeyCondition::RPNElement::FUNCTION_IS_NULL
+                || curr.function == KeyCondition::RPNElement::FUNCTION_IS_NOT_NULL;
+            const bool contains_in_set = curr.function == KeyCondition::RPNElement::FUNCTION_IN_SET
+                || curr.function == KeyCondition::RPNElement::FUNCTION_NOT_IN_SET;
+            const bool contains_in_range = curr.function == KeyCondition::RPNElement::FUNCTION_IN_RANGE
+                || curr.function == KeyCondition::RPNElement::FUNCTION_NOT_IN_RANGE;
+
+            SCOPE_EXIT({rpn_stack.pop_back();});
+
+
+            /// Key filter expressions like "func(col) > 100" are not supported for ORC filter push down
+            if (!curr.monotonic_functions_chain.empty())
+            {
+                builder.literal(orc::TruthValue::YES_NO_NULL);
+                break;
+            }
+
+            /// key filter expressions like "(a, b, c) in " or "(func(a), b) in " are not supported for ORC filter push down
+            /// Only expressions like "a in " are supported currently, maybe we can improve it later.
+            auto set_index = curr.set_index;
+            if (contains_in_set)
+            {
+                if (!set_index || set_index->getOrderedSet().size() != 1 || set_index->hasMonotonicFunctionsChain())
+                {
+                    builder.literal(orc::TruthValue::YES_NO_NULL);
+                    break;
+                }
+            }
+
+            const auto * type = getORCTypeByName(schema, getColumnNameFromKeyCondition(key_condition, curr.key_column));
+            if (!type)
+            {
+                builder.literal(orc::TruthValue::YES_NO_NULL);
+                break;
+            }
+
+            auto predicate_type = convertORCTypeToPredicateType(*type);
+            if (!predicate_type.has_value())
+            {
+                builder.literal(orc::TruthValue::YES_NO_NULL);
+                break;
+            }
+
+            if (need_wrap_not)
+                builder.startNot();
+
+            if (contains_is_null)
+            {
+                builder.isNull(type->getColumnId(), *predicate_type);
+            }
+            else if (contains_in_range)
+            {
+                const auto & range = curr.range;
+                bool has_left_bound = !range.left.isNegativeInfinity();
+                bool has_right_bound = !range.right.isPositiveInfinity();
+                if (!has_left_bound && !has_right_bound)
+                {
+                    /// Transform whole range to orc::TruthValue::YES or orc::TruthValue::YES_NULL
+                    builder.literal(range.left_included && range.right_included ? orc::TruthValue::YES : orc::TruthValue::YES_NULL);
```

Comment:
`left_included` and `right_included` are always false here, I think. So this can just be `orc::TruthValue::YES_NULL`.